core/trino-main/src/main/java/io/trino/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -1303,4 +1330,47 @@ public PlanNode getRoot()
             return mappings;
         }
     }
+
+    private static class DynamicFilterVisitor
+            extends PlanVisitor<PlanNode, Void>
```

Comment:
You can use `io.trino.sql.planner.plan.SimplePlanRewriter` instead to handle simple case